src/mentionui.js

```diff
@@ -0,0 +1,493 @@
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	destroy() {
+		super.destroy();
+
+		// Destroy created UI components as they are not automatically destroyed (see ckeditor5#1341).
+		this.panelView.destroy();
+	}
+
+	/**
+	 * Creates the {@link #panelView}.
+	 *
+	 * @private
+	 * @returns {module:ui/panel/balloon/balloonpanelview~BalloonPanelView}
+	 */
+	_creatPanelView() {
+		const panelView = new BalloonPanelView( this.editor.locale );
+
+		panelView.withArrow = false;
+		panelView.render();
+
+		this.editor.ui.view.body.add( panelView );
+
+		return panelView;
+	}
+
+	/**
+	 * Creates the {@link #_mentionsView}.
+	 *
+	 * @private
+	 * @returns {module:mention/ui/mentionsview~MentionsView}
+	 */
+	_createMentionView() {
+		const locale = this.editor.locale;
+
+		const mentionsView = new MentionsView( locale );
+
+		this._items = new Collection();
+
+		this.panelView.content.add( mentionsView );
+
+		mentionsView.listView.items.bindTo( this._items ).using( data => {
+			const { item, marker } = data;
+
+			const listItemView = new MentionListItemView( locale );
+
+			const view = this._renderItem( item, marker );
+			view.delegate( 'execute' ).to( listItemView );
+
+			listItemView.children.add( view );
+			listItemView.item = item;
+			listItemView.marker = marker;
+
+			listItemView.on( 'execute', () => {
+				mentionsView.fire( 'execute', {
+					item,
+					marker
+				} );
+			} );
+
+			return listItemView;
+		} );
+
+		mentionsView.on( 'execute', ( evt, data ) => {
+			const editor = this.editor;
+			const model = editor.model;
+
+			const item = data.item;
+			const marker = data.marker;
+
+			const watcher = this._getWatcher( marker );
+
+			const text = watcher.last;
+
+			const textMatcher = createTextMatcher( marker );
+			const matched = textMatcher( text );
+			const matchedTextLength = matched.marker.length + matched.feedText.length;
+
+			// Create a range on matched text.
+			const end = model.createPositionAt( model.document.selection.focus );
+			const start = end.getShiftedBy( -matchedTextLength );
+			const range = model.createRange( start, end );
+
+			editor.execute( 'mention', {
+				mention: item,
+				marker,
+				range
+			} );
+
+			this._hidePanel();
+		} );
+
+		return mentionsView;
+	}
+
+	/**
+	 * Returns item renderer for marker.
+	 *
+	 * @private
+	 * @param {String} marker
+	 * @returns {Function|null}
+	 */
+	_getItemRenderer( marker ) {
+		const { itemRenderer } = this._mentionsConfigurations.get( marker );
+
+		return itemRenderer;
+	}
+
+	/**
+	 * Returns a promise that resolves with autocomplete items for given text.
+	 *
+	 * @param {String} marker
+	 * @param {String} feedText
+	 * @return {Promise<module:mention/mention~MentionFeedItem>}
+	 * @private
+	 */
+	_getFeed( marker, feedText ) {
+		const { feedCallback } = this._mentionsConfigurations.get( marker );
+
+		return Promise.resolve().then( () => feedCallback( feedText ) );
+	}
+
+	/**
+	 * Registers a text watcher for marker.
```

Comment:
There's a lot going on in this method and by looking at it I suppose it's critical for the feature. Maybe a few more words about that?